src/replay/replay_recorder.cpp

```diff
@@ -116,7 +127,81 @@ void ReplayRecorder::update(int ticks)
 #ifdef DEBUG
         m_count++;
 #endif
-        if (time - m_last_saved_time[i] < stk_config->m_replay_dt)
+        // If one of the tracked kart data has significantly changed
+        // for the kart, update sooner than the usual dt
+        bool force_update = false;
+
+        int attachment;
+        if (kart->getAttachment()->getType() == Attachment::ATTACH_NOTHING)
```

Comment:
An enum already is an int, why have all that code to convert the enum to an int?

Reply:
Sure, this code is not strictly necessary at the moment, and it's not elegant either, but this is a precaution which will one day prove useful.

Why ? Because the underlying values behind "ATTACH_NOTHING", etc. can change. Say, for example, you add a new attachment (this will happen, maybe in five years, but it will). You'll put it before ATTACH_MAX and ATTACH_NOTHING ; and this will increment the value of ATTACH_NOTHING.

You won't think for a second that you've broken something by adding a new attachment type. Everything will seem to work while you'll test this attachment in normal play. It'll even work in new replays you'd record after the change. But if there is not this conversion to guarantee the independence of the stored replay values from the internal ordering (storing by meaning rather than by raw encoding), you'll have effectively broken all existing version 4+ replays, which may well be all the players replays from the last 2-3 years. If this is caught before release, probably in RC testing, you'd only have to do a code similar to the one I did here for things to keep working. Worse case scenario, this gets into release. You'd be stuck between dropping support for old replays and adding code to manage two possible encodings - the latter solution being definitely more heavy and less elegant that this code.

Note that I could have added similar code for skidding_state, but I estimated that in that case, it was extremely unlikely for the enum to change, even more twice.